sympy/solvers/diophantine/diophantine.py

```diff
@@ -3662,74 +3670,91 @@ def sum_of_three_squares(n):
     See Also
     ========
 
-    sum_of_squares()
+    power_representation :
+        ``sum_of_three_squares(n)`` is one of the solutions output by ``power_representation(n, 2, 3, zeros=True)``
+
     """
     special = {1:(1, 0, 0), 2:(1, 1, 0), 3:(1, 1, 1), 10: (1, 3, 0), 34: (3, 3, 4), 58:(3, 7, 0),
         85:(6, 7, 0), 130:(3, 11, 0), 214:(3, 6, 13), 226:(8, 9, 9), 370:(8, 9, 15),
         526:(6, 7, 21), 706:(15, 15, 16), 730:(1, 27, 0), 1414:(6, 17, 33), 1906:(13, 21, 36),
         2986: (21, 32, 39), 9634: (56, 57, 57)}
 
-    v = 0
-
+    n = as_int(n)
+    if n < 0:
+        raise ValueError("n should be a non-negative integer")
     if n == 0:
         return (0, 0, 0)
-
-    v = multiplicity(4, n)
-    n //= 4**v
-
+    n, v = remove(n, 4)
+    v = 1 << v
     if n % 8 == 7:
         return
-
-    if n in special.keys():
+    if n in special:
         x, y, z = special[n]
-        return _sorted_tuple(2**v*x, 2**v*y, 2**v*z)
+        return tuple(sorted([v*x, v*y, v*z]))
 
     s, _exact = integer_nthroot(n, 2)
-
     if _exact:
-        return (2**v*s, 0, 0)
-
-    x = None
-
+        return tuple(sorted([v*s, 0, 0]))
     if n % 8 == 3:
-        s = s if _odd(s) else s - 1
-
+        if not s % 2:
+            s -= 1
         for x in range(s, -1, -2):
             N = (n - x**2) // 2
             if isprime(N):
+                # n % 8 == 3 and x % 2 == 1 => N % 4 == 1
                 y, z = prime_as_sum_of_two_squares(N)
-                return _sorted_tuple(2**v*x, 2**v*(y + z), 2**v*abs(y - z))
-        return
-
-    if n % 8 in (2, 6):
-        s = s if _odd(s) else s - 1
-    else:
-        s = s - 1 if _odd(s) else s
+                return tuple(sorted([v*x, v*(y + z), v*abs(y - z)]))
+        # We will never reach this point because there must be a solution.
+        assert False
 
+    # assert n % 4 in [1, 2]
+    if not((n % 2) ^ (s % 2)):
+        s -= 1
     for x in range(s, -1, -2):
         N = n - x**2
         if isprime(N):
+            # assert N % 4 == 1
             y, z = prime_as_sum_of_two_squares(N)
-            return _sorted_tuple(2**v*x, 2**v*y, 2**v*z)
+            return tuple(sorted([v*x, v*y, v*z]))
+    # We will never reach this point because there must be a solution.
+    assert False
 
 
 def sum_of_four_squares(n):
     r"""
     Returns a 4-tuple `(a, b, c, d)` such that `a^2 + b^2 + c^2 + d^2 = n`.
-
     Here `a, b, c, d \geq 0`.
 
-    Usage
-    =====
+    Explanation
+    ===========
+
+    If there exists a 3-tuple `(a, b, c)` in `n` such that `n = a^2 + b^2 + c^2`, then return a 4-tuple `(0, a, b, c)`.
+    If there exists a 3-tuple `(a, b, c)` in `n-1` such that `n-1 = a^2 + b^2 + c^2`, then return a 4-tuple `(1, a, b, c)`.
```

Comment:
```suggestion
```suggestion
    If there is no solution having all values greater than 1 then if there exists a 3-tuple
    `(a, b, c)` in `n` such that `n = a^2 + b^2 + c^2`, then return a 4-tuple `(0, a, b, c)`; and
    if there exists a 3-tuple `(a, b, c)` in `n-1` such that `n-1 = a^2 + b^2 + c^2`, then return a 4-tuple `(1, a, b, c)`.
```

Reply:
That would be my preference for how it performs, else how do you reframe the request to get (8, 8, 32, 48) instead of (0, 8, 16, 56)?